Add remote repository to settings

isc-etamarch opened the pull request:
Fixes #448
Fixes #375

isc-etamarch committed:
Added remote repository to settings

isc-etamarch committed:
Merge branch 'main' into issue-448

isc-tleavitt commented on csp/gitprojectsettings.csp:
```diff
@@ -94,6 +96,11 @@ body {
         }
 
+        set newRemote = $Get(%request.Data("remoteRepo",1))
+        if (newRemote '= ##class(SourceControl.Git.Utils).GetConfiguredRemote()) {
```
This looks like it's going to replace an uncensored remote with a censored remote, breaking things when you change the settings on HCC.

isc-tleavitt reviewed (changes requested):
This needs a CHANGELOG.md entry

isc-etamarch committed:
Update changelog, fix updating remote

Added changelog entry
Check censored remote against censored remote to see if the settings entry has been changed

isc-tleavitt commented on cls/SourceControl/Git/Utils.cls:
```diff
@@ -2798,4 +2798,28 @@ ClassMethod BaselineExport(pCommitMessage = "", pPushToRemote = "") As %Status
+    return url
+}
+
+ClassMethod GetCensoredRemote() As %String
```
I'm going to be pedantic and say we should call this "GetRedactedRemote" instead, here...

isc-tleavitt commented on csp/gitprojectsettings.csp:
```diff
@@ -326,6 +334,16 @@ body {
+            <div class="col-sm-7">
+                <input type="text" class="form-control" id="remoteRepo" name="remoteRepo" value='#(..EscapeHTML(remote))#' placeholder="ex. git@github.com:User/UserRepo.git"/>
+                <div class = "neutral-feedback">
+                    (Username is censored)
```
... and also here; say "Credentials are redacted"

isc-tleavitt reviewed (changes requested):
Sorry... one more thing I think would be good to change. Otherwise good to go.

isc-etamarch committed:
Replace 'censored' with 'redacted'

isc-etamarch committed:
Merge branch 'main' into issue-448
